src/docs/timestamp-txn.dox

```diff
@@ -74,7 +74,7 @@ transaction commits, using WT_SESSION::commit_transaction:
 | Timestamp | Constraint | Description |
 |-----------|------------|-------------|
 | commit_timestamp | >= stable and >= prepare and >= any system read timestamp | see @ref timestamp_txn_api_commit_timestamp; the transaction's commit timestamp |
```

Comment:
Unrelated to this PR. The commit timestamp also must be greater than the stable timestamp. We need to fix this scenario also. Maybe in a different PR.

Reply:
That should probably be another ticket, yeah. (It should be > stable and >= prepare and > any system read timestamp, right?)

Reply:
Ah I see the confusion. I retract my statement regarding this specific line, but I want to be clear that the usage of commit_timestamp is unfortunately ambiguous. 

This is the old vestige that back in 4.0 (MDB versioning terms) there was only the commit_timestamp. Today when a `timestamp_transaction` may choose to only supply a `commit_timestamp`, and the `durable_timestamp` will default to the `commit_timestamp`. I think if we had prepared machinery in at the same time as introducing timestamps, we would have made `durable_timestamp` required in the input to `timestamp_transaction` and `commit_timestamp` was optional/only required for prepared transactions.

So I find it more natural to normalize the language and refer to variables after we apply whichever defaults. I would argue that the durable_timestamp must always be greater than the stable timestamp. And by a transitive property, for non-prepared transactions, the commit_timestamp must also be greater than the stable timestamp.

Hopefully we can avoid confusion in these tables that when a durable_timestamp and commit_timestamp exist, the commit timestamp does not need to be related to the stable timestamp.

Reply:
Right, the constraints are different for prepared and non-prepared transactions. The issue here is that the restriction is off by one, at least in the docs and according to WT-7712 also in the code.